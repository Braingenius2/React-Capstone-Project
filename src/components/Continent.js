import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Continent = ({ map, name, covidCases }) => {
  const continentName = name.replace(/\s/g, '-');

  return (
    <NavLink to={`${continentName}`}>
      <div>
        <div>arrow right img here</div>
      </div>
      <div>
        <img src={map} alt={name} />
      </div>
      <div>
        <h3>{ name }</h3>
        <p>
          { covidCases }
          {' '}
          Covid cases
        </p>
      </div>
    </NavLink>
  );
};

Continent.propTypes = {
  map: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  covidCases: PropTypes.string.isRequired,
};

export default Continent;
