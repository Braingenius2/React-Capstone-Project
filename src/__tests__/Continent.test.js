import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, screen } from '@testing-library/react';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import continentMaps from '../components/continentsMaps';
import Continent from '../components/Continent';

describe('The continent component', () => {
  test('should render correctly into the DOM', () => {
    const { container } = render(
      <MemoryRouter>
        <Continent map={continentMaps.Africa} name="Africa" covidCases="1000" index= '3' columnIndex='3' />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should navigate to the second page when user clicked on Australia-Oceania continent item', () => {
    render(
      <BrowserRouter>
        <Continent map={continentMaps['Australia-Oceania']} name="Australia-Oceania" covidCases="1,345,890" index= '3' columnIndex='3' />
      </BrowserRouter>,
    );
    const continentLink = document.querySelector('.continent-item');
    fireEvent.click(continentLink);
    expect(window.location.pathname).toBe('/Australia-Oceania');
  });

  test('should render the text: Europe and the number of cases', () => {
    render(
      <MemoryRouter>
        <Continent map={continentMaps.Europe} name="Europe" covidCases="20,105,600" index= '3' columnIndex='3' />
      </MemoryRouter>,
    );
    const continentTitle = screen.queryByText(/Europe/i);
    const continentCases = screen.queryByText(/20,105,600 cases/i);

    expect(continentTitle).toBeInTheDocument();
    expect(continentCases).toBeInTheDocument();
  });
});
