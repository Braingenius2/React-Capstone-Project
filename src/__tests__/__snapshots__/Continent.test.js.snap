// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The continent component should render correctly into the DOM 1`] = `
<div>
  <a
    class="p-4 bg-pink-500"
    href="/Africa"
  >
    <div>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <polyline
          points="12 16 16 12 12 8"
        />
        <line
          x1="8"
          x2="16"
          y1="12"
          y2="12"
        />
      </svg>
    </div>
    <div>
      <img
        alt="Africa"
        src="Africa-Map.png"
        style="filter: opacity(0.7);"
      />
    </div>
    <div>
      <h3
        class="text-3xl font-bold"
      >
        Africa
      </h3>
      <p>
        1000
         
        Covid cases
      </p>
    </div>
  </a>
</div>
`;
